consensus/consensus_test.go

```diff
@@ -0,0 +1,159 @@
+package consensus
+
+// consensus_test.go contains objects and functions that help to build an
+// architecture for testing. Such objects include a very basic wallet and a
+// basic miner.
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+)
+
+// An assistant keeps track of addresses and contracts and whatnot to help with
+// testing. There are also helper functions for mining blocks and cobbling
+// together transactions. It's designed to be simple, and it's not very smart
+// or efficient.
+type assistant struct {
+	state  *State
+	tester *testing.T
+
+	unlockConditions UnlockConditions
+	unlockHash       UnlockHash
+	secretKey        crypto.SecretKey
+}
```

Comment:
I'm not planning on fixing all of the testing in this PR. the diffs are plenty large enough :p